langchain4j-weaviate/src/main/java/dev/langchain4j/store/embedding/WeaviateEmbeddingStoreImpl.java

```diff
@@ -0,0 +1,192 @@
+package dev.langchain4j.store.embedding;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+
+import dev.langchain4j.data.embedding.Embedding;
+import dev.langchain4j.data.segment.TextSegment;
+import io.weaviate.client.Config;
+import io.weaviate.client.WeaviateAuthClient;
+import io.weaviate.client.WeaviateClient;
+import io.weaviate.client.v1.data.model.WeaviateObject;
+import io.weaviate.client.v1.data.replication.model.ConsistencyLevel;
+import io.weaviate.client.v1.graphql.query.argument.NearVectorArgument;
+import io.weaviate.client.v1.graphql.query.fields.Field;
+import java.security.MessageDigest;
+import java.util.*;
+import java.util.stream.Collectors;
+import lombok.Builder;
+import lombok.SneakyThrows;
+import lombok.val;
+
+public class WeaviateEmbeddingStoreImpl implements EmbeddingStore<TextSegment> {
+
+  private static final String DEFAULT_CLASS = "Default";
+  private static final Double DEFAULT_MIN_CERTAINTY = 0.8;
+  private static final String METADATA_TEXT_SEGMENT = "text";
+  private static final String ADDITIONALS = "_additional";
+
+  private final WeaviateClient client;
+  private final String objectClass;
+
+  @Builder
+  @SneakyThrows
+  public WeaviateEmbeddingStoreImpl(String apiKey, String scheme, String host, String objectClass) {
+    client = WeaviateAuthClient.apiKey(new Config(scheme, host), apiKey);
+    this.objectClass = objectClass != null ? objectClass : DEFAULT_CLASS;
+  }
+
+  @Override
+  public String add(Embedding embedding) {
+    val id = generateRandomId();
+    add(id, embedding);
+    return id;
+  }
+
+  @Override
+  public void add(String id, Embedding embedding) {
```

Comment:
It seems that this method does not work. I tried this:
```java
Embedding embedding = embeddingModel.embed("test");
embeddingStore.add("12345", embedding);
List<EmbeddingMatch<TextSegment>> relevant = embeddingStore.findRelevant(embedding, 1000);
```
And I dont get my embedding back. When using `embeddingStore.add(embedding);`, it works. Does weaviate expects ID to always be UUID?

Reply:
yes, ID is in Weaviate UUID, e.g. https://weaviate.io/developers/weaviate/manage-data/create

Reply:
Could you please than add a javadoc to `WeaviateEmbeddingStore` for methods that accept ID/IDs mentioning that ID should be strictly in UUID format, with a link for more details (https://en.wikipedia.org/wiki/Universally_unique_identifier)? Thanks!